File: progra3-0583-equipo5-frontend/Frontend_C#/SoftPac/SoftPacBusiness/AcreedoresBO.cs
// SoftPac.Business.AcreedoresBO.cs
using SoftPac.Persistance.DAOImpl;
using SoftPac.Model;
using System;
using System.ComponentModel;

namespace SoftPac.Business
{
    public class AcreedoresBO
    {
        private AcreedoresDAOImpl acreedorDAO;

        public AcreedoresBO()
        {
            this.acreedorDAO = new AcreedoresDAOImpl();
        }

        public int insertar(string razon_social, string ruc, string direccion_fiscal,
                            string condicion, int plazo_de_pago, string activo, int id_pais)
        {
            AcreedoresDTO dto = new AcreedoresDTO();
            dto.RazonSocial = razon_social;
            dto.Ruc = ruc;
            dto.DireccionFiscal = direccion_fiscal;
            dto.Condicion = condicion;
            dto.PlazoDePago = plazo_de_pago;
            dto.Activo = activo == "S";
            PaisesDTO pais = new PaisesDTO();
            pais.PaisId = id_pais;
            dto.Pais = pais;

            return this.acreedorDAO.insertar(dto);
        }

        public int modificar(int id_acreedor, string razon_social, string ruc,
                             string direccion_fiscal, string condicion, int plazo_de_pago, string activo,
                             int id_pais)
        {
            AcreedoresDTO dto = new AcreedoresDTO();
            dto.AcreedorId = id_acreedor;
            dto.RazonSocial = razon_social;
            dto.Ruc = ruc;
            dto.DireccionFiscal = direccion_fiscal;
            dto.Condicion = condicion;
            dto.PlazoDePago = plazo_de_pago;
            dto.Activo = activo == "S";
            PaisesDTO pais = new PaisesDTO();
            pais.PaisId = id_pais;
            dto.Pais = pais;

            return this.acreedorDAO.modificar(dto);
        }

        public int Eliminar(int acreedorId, UsuariosDTO usuario)
        {
            AcreedoresDTO dto = new AcreedoresDTO();
            dto.AcreedorId = acreedorId;
            dto.UsuarioEliminacion = usuario;
            dto.FechaEliminacion = DateTime.Now;
            return this.acreedorDAO.eliminarLogico(dto);
        }

        public int Eliminar(AcreedoresDTO acreedor)
        {
            return this.acreedorDAO.eliminarLogico(acreedor);
        }

        public AcreedoresDTO obtenerPorId(int acreedor_id)
        {
            return this.acreedorDAO.obtenerPorId(acreedor_id);
        }

        public BindingList<AcreedoresDTO> ListarTodos()
        {
            return (BindingList<AcreedoresDTO>)this.acreedorDAO.ListarTodos();
        }
    }
}

File: BackEnd/Backend_Java/SoftPacBusiness/src/main/java/pe/edu/pucp/softpac/bo/AcreedoresBO.java

package pe.edu.pucp.softpac.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softpac.dao.AcreedoresDAO;
import pe.edu.pucp.softpac.daoImpl.AcreedoresDAOImpl;
import pe.edu.pucp.softpac.model.AcreedoresDTO;
import pe.edu.pucp.softpac.model.PaisesDTO;
import pe.edu.pucp.softpac.model.UsuariosDTO;


public class AcreedoresBO {
    
   private AcreedoresDAO acreedorDAO;
   
   public AcreedoresBO(){
       this.acreedorDAO = new AcreedoresDAOImpl();
   }
   
   public Integer insertar(String razon_social, String ruc, String direccion_fiscal,
            String condicion, Integer plazo_de_pago, String activo, Integer id_pais){
       AcreedoresDTO acreedoresDTO = new AcreedoresDTO();
       acreedoresDTO.setRazon_social(razon_social);
       acreedoresDTO.setRuc(ruc);
       acreedoresDTO.setPlazo_de_pago(plazo_de_pago);
       acreedoresDTO.setCondicion(condicion);
       acreedoresDTO.setDireccion_fiscal(direccion_fiscal);
       acreedoresDTO.setActivo(activo.equals("S") ? Boolean.TRUE : Boolean.FALSE);
       PaisesDTO pais = new PaisesDTO();
       pais.setPais_id(id_pais);
       acreedoresDTO.setPais(pais);
            
       return this.acreedorDAO.insertar(acreedoresDTO);
   }
   
   public Integer modificar(Integer id_acreedor, String razon_social, String ruc,
           String direccion_fiscal, String condicion, Integer plazo_de_pago, String activo,
           Integer id_pais){
       
        // Validaciones
        if (id_acreedor == null || id_acreedor <= 0) {
            throw new IllegalArgumentException("El ID del acreedor es obligatorio");
        }
       
       AcreedoresDTO acreedoresDTO = new AcreedoresDTO();
       acreedoresDTO.setAcreedor_id(id_acreedor);
       acreedoresDTO.setRazon_social(razon_social);
       acreedoresDTO.setRuc(ruc);
       acreedoresDTO.setDireccion_fiscal(direccion_fiscal);
       acreedoresDTO.setCondicion(condicion);
       acreedoresDTO.setPlazo_de_pago(plazo_de_pago);
       acreedoresDTO.setActivo(activo.equals("S") ? Boolean.TRUE : Boolean.FALSE);
       PaisesDTO pais = new PaisesDTO();
       pais.setPais_id(id_pais);
       acreedoresDTO.setPais(pais);
       return this.acreedorDAO.modificar(acreedoresDTO);
   }
   
   public Integer eliminarLogico(AcreedoresDTO acreedor, UsuariosDTO usuarioActual){
       
       if (acreedor == null || acreedor.getAcreedor_id() == null) {
           throw new IllegalArgumentException("El acreedor no existe o no tiene ID válido.");
       }

       acreedor.setFecha_eliminacion(new Date());
       acreedor.setUsuario_eliminacion(usuarioActual);
       
       return acreedorDAO.eliminarLogico(acreedor);
   }
   
   
    public AcreedoresDTO obtenerPorId(Integer acreedor_id){
       AcreedoresDTO acreedores = new AcreedoresDTO();
       acreedores.setAcreedor_id(acreedor_id);
       return this.acreedorDAO.obtenerPorId(acreedor_id);

    }
    
    public ArrayList<AcreedoresDTO> listarTodos(){
        return (ArrayList<AcreedoresDTO>) acreedorDAO.listarTodos();
    }
    
    
}
